### PR TITLE
Add new runner labels to actionlint

### PR DESCRIPTION
**Pull request by ZainRizvi**
Adding the labels corresponding to the Amazon2023 ami

### CODE REVIEW

**Review by atalman — APPROVED** [review]

File: .github/actionlint.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ self-hosted-runner:
     - linux.8xlarge.nvidia.gpu
     - linux.16xlarge.nvidia.gpu
     - linux.g5.4xlarge.nvidia.gpu
-    # Organization-wide AWS Linux Runners on Linux Foundation account
+    # Pytorch/pytorch AWS Linux Runners on Linux Foundation account
```
Comment (atalman) [inline .github/actionlint.yaml:19]: These can't be used to run domain jobs ?

File: .github/actionlint.yaml [hunk 2]
```diff
@@ -27,6 +27,28 @@ self-hosted-runner:
     - lf.linux.8xlarge.nvidia.gpu
     - lf.linux.16xlarge.nvidia.gpu
     - lf.linux.g5.4xlarge.nvidia.gpu
+    # Organization-wide AWS Linux Runners with new Amazon 2023 AMI
+    - amz2023.linux.large
```
Comment (jeanschmidt) [inline .github/actionlint.yaml:31]: It starting to become uncomfortable to double all runners.

Can we come up with a more programatic approach? Like make the lint underrstand that those runners can have the suffix `amz2023.`?

File: .github/actionlint.yaml (hunk 1 above)
Comment (ZainRizvi) [reply .github/actionlint.yaml:19]: Nope

File: .github/actionlint.yaml (hunk 2 above)
Comment (ZainRizvi) [reply .github/actionlint.yaml:31]: Yeah, this is at the limits of acceptability. If we have to double these again, then we'll want to see how to make this more scalable (maybe use codegen to make this file or something).

**Pull request closed without merging**